plugins/polio/tasks/api/refresh_lqas_data.py

```diff
@@ -42,34 +33,40 @@ def validate(self, attrs):
         return validated_data
 
 
-class ExternalTaskSerializer(TaskSerializer):
-    def update(self, task, validated_data):
-        has_progress_message = validated_data.get("progress_message", None) is not None
-        has_status = validated_data.get("status", None) is not None
-        has_progress_value = validated_data.get("progress_value", None) is not None
-        has_end_value = validated_data.get("end_value", None) is not None
-        if (has_status or has_progress_value or has_progress_message or has_end_value) and not task.external:
-            raise serializers.ValidationError({"external": "Cannot modify non external tasks"})
-        if validated_data.get("should_be_killed", None) is not None:
-            task.should_be_killed = validated_data["should_be_killed"]
-            if validated_data["should_be_killed"]:
-                task.status = KILLED
-        if has_status:
-            task.status = validated_data["status"]
-            if (
-                validated_data["status"] == SUCCESS
-                or validated_data["status"] == ERRORED
-                or validated_data["status"] == KILLED
-            ):
-                task.ended_at = datetime.now()
-        if has_progress_message:
-            task.progress_message = validated_data["progress_message"]
-        if has_progress_value:
-            task.progress_value = validated_data["progress_value"]
-        if has_end_value:
-            task.end_value = validated_data["end_value"]
-        task.save()
-        return task
+class RefreshLQASDataPostSerializer(ExternalTaskPostSerializer):
+    def validate(self, attrs):
+        validated_data = super().validate(attrs)
+        request = self.context["request"]
+        slug = validated_data.get("slug", None)
+        config = validated_data.get("config", None)
+        id_field = validated_data.get("id_field", None)
+        error = {}
+        if slug is None or slug != LQAS_CONFIG_SLUG:
+            error["slug"] = f"Wrong config slug. Expected {LQAS_CONFIG_SLUG}, got {slug}"
+        if config is None:
+            error["config"] = "This field is mandatory"
+        if id_field is None:
+            error["id_field"] = "This field is mandatory"
```

Comment:
Aren't these checks already done by `ExternalTaskPostSerializer` ?

Reply:
Only for `slug` really. `config` and `id_field` are not mandatory, as their value depend on the OpenHexa pipeline. 
I added stricter checks here, which are specific to the lqas pipeline